scripts/serving/wrapper/preprocessor.py

```diff
@@ -102,19 +122,30 @@ def start(self):
         ormb_file_path = os.path.join(
             self.model_root_path, self._serving_name, "ormbfile.yaml")
         if not os.path.exists(ormb_file_path):
+            logger.error(f'{ormb_file_path} does not exist')
             return
 
         # Phase 1: Extract model_format and yaml
         yaml_data = self._extract_yaml()
+        if 'format' in yaml_data.items():
+            logger.error('model format missing')
+            return
         format = yaml_data["format"]
+        # set env for mlserver
+        os.putenv('MODEL_FORMAT', format)
+
+        if 'version' in yaml_data.items():
+            version = yaml_data["version"]
+        else:
+            version = 'v1.0.0'
 
         # Phase 2: Generate 'config.pbtxt' if need
-        if format != 'PMML' and format != 'SKLearn' and format != 'XGBoost':
+        if format != 'PMML' and format not in Preprocessor.mlserver_model:
             self._generate_config_pbtxt(yaml_data)
 
         # Phase 3: Generate 'model setting' if need
-        if format == 'SKLearn' or format == 'XGBoost':
-            self._generate_model_setting(format)
+        if format in Preprocessor.mlserver_model:
+            self._generate_model_setting(format, version)
 
```

Comment:
Can we rewrite these codes for future scalability ?

Reply:
This is awkward... I refactored this in another PR.